src/libmongoc/src/mongoc/mongoc-openssl.c

```diff
@@ -84,6 +86,14 @@ _mongoc_openssl_init (void)
       MONGOC_ERROR ("Failed to initialize OpenSSL.");
    }
 
+#ifdef NID_tlsfeature
+   tlsfeature_nid = NID_tlsfeature;
+#else
+   /* TLS versions before 1.1.0 did not define the TLS Feature extension. */
+   tlsfeature_nid =
+      OBJ_create ("1.3.6.1.5.5.7.1.24", "tlsfeature", "TLS Feature");
```

Comment:
The server does something similar to define the OID when it is not predefined in OpenSSL:
https://github.com/mongodb/mongo/blob/f627ff829ba96487fccb6945d0c2807b1bb51d52/src/mongo/util/net/ssl_manager_openssl.cpp#L357-L361